### PR TITLE
Add test cases for EnumerableGuardianMap.

### PR DESCRIPTION
**Pull request by wshino**

### CODE REVIEW

File: test/unit/libraries/EnumerableGuardianMap/set.t.sol [hunk 1]
```diff
@@ -2,16 +2,109 @@
...
+        require(
+            guardiansStorage[vm.addr(1)]._values[guardian1].status ==
+                GuardianStatus.ACCEPTED,
+            "Expected status to be ACCEPTED"
+        );
+    }
+
+    function test_Set_RevertWhen_MaxNumberOfGuardiansReached() public {
+        bool result;
+        // TODO: Can it be acceptable to number 33?
```
Comment (wshino) [inline test/unit/libraries/EnumerableGuardianMap/set.t.sol]: Regarding current implementation, it allows to store 33 guardians. Is that expected behavior?

Comment (JohnGuilding) [reply test/unit/libraries/EnumerableGuardianMap/set.t.sol]: Ah yes good catch, expected behaviour should be 32. Could you change to assertion in the library to ensure that the max number is 32?

Comment (wshino) [reply test/unit/libraries/EnumerableGuardianMap/set.t.sol]: @JohnGuilding Fixed.

File: test/unit/libraries/EnumerableGuardianMap/keys.t.sol [hunk 2]
```diff
@@ -2,23 +2,72 @@
...
         super.setUp();
     }
 
     function test_Keys_StartsEmpty() public view {
-        // TODO: test
+        address[] memory keys = guardiansStorage[accountAddress].keys();
+        assertEq(keys.length, 0);
     }
+
     function test_Keys_ReturnsEmptyArrayOfKeys() public view {
```
Comment (JohnGuilding) [inline test/unit/libraries/EnumerableGuardianMap/keys.t.sol]: cool, can you remove this test case then?

File: test/unit/libraries/EnumerableGuardianMap/removeAll.t.sol [hunk 3]
```diff
@@ -2,21 +2,82 @@
...
+                    guardianWeights[1]
+                )
+            });
+            assertEq(result, true);
+        }
+        address[] memory addresses = new address[](3);
+        addresses[0] = vm.addr(1);
+        addresses[1] = vm.addr(2);
+        addresses[2] = vm.addr(3);
+        guardiansStorage[accountAddress].removeAll(addresses);
```
Comment (JohnGuilding) [inline test/unit/libraries/EnumerableGuardianMap/removeAll.t.sol:34]: Can you add an assert to check that the values are no longer in storage?

File: test/unit/libraries/EnumerableGuardianMap/removeAll.t.sol [hunk 4]
```diff
@@ -2,21 +2,82 @@
...
+            EnumerableGuardianMap.MAX_NUMBER_OF_GUARDIANS
+        );
+        for (
+            uint256 i = 0;
+            i < EnumerableGuardianMap.MAX_NUMBER_OF_GUARDIANS;
+            i++
+        ) {
+            addresses[i] = vm.addr(i + 1);
+        }
+        guardiansStorage[accountAddress].removeAll(addresses);
```
Comment (JohnGuilding) [inline test/unit/libraries/EnumerableGuardianMap/removeAll.t.sol:72]: Same here, can you add an assert to check that the values are no longer in storage?

File: test/unit/libraries/EnumerableGuardianMap/set.t.sol [hunk 5]
```diff
@@ -2,16 +2,108 @@
...
-    function test_Set_ReturnsFalseWhen_AddingKeysAlreadyInTheSet() public view { }
-    function test_Set_UpdatesValuesForKeysAlreadyInTheSet() public view { }
-    function test_Set_RevertWhen_MaxNumberOfGuardiansReached() public view { }
-    function test_Set_AddsAKey() public view { }
-    function test_Set_AddsSeveralKeys() public view { }
+    function test_Set_AddsAKey() public {
+        guardiansStorage[accountAddress].set({
+            key: guardian1,
+            value: GuardianStorage(GuardianStatus.REQUESTED, guardianWeights[1])
+        });
```
Comment (JohnGuilding) [inline test/unit/libraries/EnumerableGuardianMap/set.t.sol:21]: Can you add an assert here

File: test/unit/EmailRecoveryManager/getRecoveryRequest.t.sol [hunk 6]
```diff
@@ -2,17 +2,29 @@
...
+        handleRecovery(recoveryModuleAddress, calldataHash, accountSalt1);
+
+        IEmailRecoveryManager.RecoveryRequest
+            memory recoveryRequest = emailRecoveryManager.getRecoveryRequest(
+                accountAddress
+            );
+        assertEq(recoveryRequest.executeAfter, 0);
+        assertEq(recoveryRequest.executeBefore, 0);
+        assertEq(recoveryRequest.currentWeight, 1);
+        assertEq(recoveryRequest.calldataHashString, "");
```
Comment (JohnGuilding) [inline test/unit/EmailRecoveryManager/getRecoveryRequest.t.sol]: If you merge the latest changes in, this has been changed to bytes32 from a string

**Review by JohnGuilding — APPROVED** [review]
Looks great, thanks for getting these done. Left some comments. 

Before merging into main will need to merge the latest changes as there are some merge conflicts.

**Review by JohnGuilding — APPROVED** [review]
LGTM

**Pull request merged**